app.js

```diff
@@ -3,16 +3,26 @@ const app = express();
 app.use (express.json()) ;
 const {
     getTopics,
-    addToApi
+    getData
 } = require('./controllers/controller.js');
 
 
 app.get('/api/topics', getTopics);
-app.get('/api', addToApi);
+app.get('/api', getData);
+
+app.all('/*', (req, res, next) => {
+  res.status(404).send({msg: "Invalid api endpoint"});
+});
 
 app.use((err, req, res, next) => {
     console.error(err);
     res.status(500).send({ msg: 'Internal Server Error' });
   });
   
+  app.use((err, req, res, next) => {
+    if (err === "Invalid query") {
```

Comment:
We don't have to worry about queries for this endpoint as none are accepted. The user can only go to /api if they go to a non-existing endpoint, it should hit your 404 error handler. You're very close here in how we will handle errors later though 😄